enterprise_access/apps/provisioning/api.py

```diff
@@ -108,3 +109,60 @@ def get_or_create_enterprise_catalog(enterprise_customer_uuid, catalog_title, ca
     )
     logger.info('Provisioning: created enterprise catalog with uuid %s', created_catalog.get('uuid'))
     return created_catalog
+
+
+def get_or_create_customer_agreement(enterprise_customer_uuid, customer_slug, default_catalog_uuid=None, **kwargs):
+    """
+    Get or create a customer agreement record from the license-manager service.
+    """
+    client = LicenseManagerApiClient()
+    existing_agreement = client.get_customer_agreement(enterprise_customer_uuid)
+    if existing_agreement:
+        logger.info('Provisioning: customer agreement with uuid %s already_exists', existing_agreement.get('uuid'))
+        return existing_agreement
+
+    created_agreement = client.create_customer_agreement(
+        enterprise_customer_uuid,
+        customer_slug,
+        default_catalog_uuid=default_catalog_uuid,
+        **kwargs,
+    )
+    logger.info('Provisioning: created customer agreement with uuid %s', created_agreement.get('uuid'))
+    return created_agreement
+
+
+def get_or_create_subscription_plan(
+    customer_agreement_dict, plan_title, catalog_uuid, opp_line_item,
+    start_date, expiration_date, desired_num_licenses, **kwargs
+):
+    """
+    Get or create a new subscription plan, provided an existing customer agreement dictionary.
+    """
+    existing_subscriptions = customer_agreement_dict.get('subscriptions', [])
+    matching_subscription = next((
+        _sub for _sub in existing_subscriptions
+        if _sub.get('salesforce_opportunity_line_item') == opp_line_item
+    ), None)
+    if matching_subscription:
+        logger.info(
+            'Provisioning: subscription plan with uuid %s and salesforce_opportunity_line_item %s already exists',
+            matching_subscription['uuid'], matching_subscription['salesforce_opportunity_line_item']
+        )
+        return matching_subscription
+
+    client = LicenseManagerApiClient()
+    created_subscription = client.create_subscription_plan(
```

Comment:
[consideration] I'm wondering if it might make sense to include the the subscription plan product type (e.g., Trial vs. B2B Paid), versus relying on the pass-thru `kwargs`.

Reply:
Related, I'm curious if we've considered whether we might want to "infer" the appropriate catalog uuid based on this product type (e.g., all Trials get a specific catalog with a subset of courses, all B2B Paid get the standard subscriptions catalog). While this secondary question probably doesn't have immediate implications for this Python API method, it might inform the caller of `get_or_create_subscription_plan`.

More concretely, is the intent that the Salesforce integration calling the provisioning API provides the catalog uuid itself, or is the catalog identifier abstracted away from Salesforce, where Salesforce should be more about which product type is sent.